### PR TITLE
Deterministic output for end users

### PR DESCRIPTION
**Pull request by NikitaAvvakumov**
https://github.com/stympy/faker/pull/207

Needed to modify all usages of `sample`, `shuffle` & `rand`. Others? (Also might have missed some.)

Will work on docs next.

Usage:

``` ruby
2.2.0 :001 > require_relative 'lib/factory-helper'
true
2.2.0 :002 > FactoryHelper::Config.seed = 99 # => Random.new(99)
99
2.2.0 :003 > FactoryHelper::Address.city
"South Greyson"
2.2.0 :004 > FactoryHelper::Address.city
"East Lacyshire"
2.2.0 :005 > FactoryHelper::Address.city
"Lake Madilynton"
2.2.0 :006 > FactoryHelper::Config.seed = 99 # => Random.new(99)
99
2.2.0 :007 > FactoryHelper::Address.city
"South Greyson"
2.2.0 :008 > FactoryHelper::Address.city
"East Lacyshire"
2.2.0 :009 > FactoryHelper::Config.seed = nil # => Random.new
nil
2.2.0 :010 > FactoryHelper::Address.city
"Dorothychester"
2.2.0 :011 > FactoryHelper::Address.city
"East Chesley"
2.2.0 :012 > FactoryHelper::Address.city
"Aliastad"
```

### CODE REVIEW

**Comment by NikitaAvvakumov** [comment]
Updated. Test iterates over all of gem's public methods to demonstrate reproducible output.

Provided default arguments for all methods that were lacking them, reduced public API of `Internet` class for the sake of testing and as general good practice, refactored `Bitcoin` to use seed-able `Random` instead of `SecureRandom`.

File: lib/factory-helper.rb [hunk 1]
```diff
@@ -17,12 +17,21 @@
...
     class << self
       attr_writer :locale
+      attr_reader :random
+
       def locale
         @locale || I18n.locale
       end
+
+      def seed=(seed)
+        @seed = seed
```
Comment (b264) [inline lib/factory-helper.rb]: I'm not sure we need to store `@seed`.  `FactoryHelper::Config.random.seed` should accomplish the same goal while also accessing the seed used if none was explicitly provided.

File: lib/factory-helper/team.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ def creature
       end
 
       def state
-        fetch('factory_helper.address.state').titleize
+        fetch('address.state').split(' ').map(&:capitalize).join(' ')
```
Comment (NikitaAvvakumov) [inline lib/factory-helper/team.rb:15]: This ain't Rails :smile: The Team subclass overall is not well implemented and is mostly untested. Added task to Trello.

File: lib/factory-helper/date.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ class Date < Base
 
     class << self
 
-      def between(from, to)
+      def between(from=::Date.today-10, to=::Date.today+10)
```
Comment (NikitaAvvakumov) [inline lib/factory-helper/date.rb:6]: I didn't give much thought to the default values, here and elsewhere, just put something in to make tests pass. Input welcome.

File: lib/factory-helper/string.rb [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
+        if length.class.included_modules.include? Enumerable
+          utf8string length.to_enum.to_a.sample(:random => FactoryHelper::Config.random)
+        else
+          utf8string length
+        end
       end
-    end
 
-  protected
+      private
```
Comment (NikitaAvvakumov) [inline lib/factory-helper/string.rb:12]: Private methods aren't listed in `object.methods`, which works better for not calling them programmatically. If protected is really preferred, can work with that too.

File: spec/factory_helper/string_spec.rb [hunk 5]
```diff
@@ -1,5 +1,3 @@
-require 'spec_helper'
```
Comment (NikitaAvvakumov) [inline spec/factory_helper/string_spec.rb:1]: Included in .rspec

**Comment by NikitaAvvakumov** [comment]
README updated, [WIP] removed

File: README.md [hunk 6]
```diff
@@ -468,7 +491,7 @@ See [CONTRIBUTING.md](https://github.com/factory-helper/factory-helper/blob/mast
 
 Contact
 -------
-Comments and feedback are welcome. Send an email to factory-helper@googlegroups.com.
+Comments and feedback are welcome via GitHub's issues.
```
Comment (b264) [inline README.md:494]: :+1:

File: lib/factory-helper/string.rb (hunk 4 above)
Comment (b264) [inline lib/factory-helper/string.rb:12]: I think the use-case here perfectly fits a protected method, but I agree it's better not to be listed in `object.methods`.

File: lib/factory-helper/team.rb (hunk 2 above)
Comment (b264) [inline lib/factory-helper/team.rb:15]: :+1:

File: lib/factory-helper/internet.rb [hunk 7]
```diff
@@ -104,13 +104,28 @@ def url(host = domain_name, path = "/#{user_name}")
...
         (words || FactoryHelper::Lorem::words(2).join(' ')).gsub(' ', glue).downcase
       end
 
       def device_token
-        rand(16 ** 64).to_s(16).rjust(64, '0').chars.to_a.shuffle.join
+        FactoryHelper::Config.random.rand(16 ** 64).to_s(16).rjust(64, '0').
+          chars.to_a.shuffle(:random => FactoryHelper::Config.random).join
       end
+
+      private
```
Comment (b264) [inline lib/factory-helper/internet.rb:116]: This breaks backwards-compatibility

Comment (NikitaAvvakumov) [inline lib/factory-helper/internet.rb:116]: Didn't realize it was documented as part of the public API. Can't imagine people using it, but hey...
An alternative is to give it default args, which seems like a silly idea, but doesn't break backwards compatibility.

Comment (b264) [inline lib/factory-helper/internet.rb:116]: I've added a test for this

**Comment by b264** [comment]
Superceded; https://github.com/factory-helper/factory-helper/pull/36

**Pull request closed without merging**